src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -250,20 +256,33 @@ public DarCollection getByCollectionId(Integer collectionId) {
     if (Objects.isNull(collection)) {
       throw new NotFoundException("Collection with the collection id of " + collectionId + " was not found");
     }
-    List<DarCollection> populatedCollections = addDatasetsToCollections(Collections.singletonList(collection));
+    List<DarCollection> populatedCollections = addDatasetsToCollections(Collections.singletonList(collection), List.of());
     return populatedCollections.stream().findFirst().orElse(null);
   }
 
-  public List<DarCollection> addDatasetsToCollections(List<DarCollection> collections) {
-
+  // This method could be a lot simpler if we passed a list of IDs to filter instead of full datasets.
+  // If there are any ids, then this method will filter the Collection Datasets by those ids.
+  // What that lets you do is stop checking for null and also to not explicitly pass nulls as arguments to this method.
```

Comment:
nit: I don't think we need this comment here

Reply:
Sorry, didn't mean to leave those in, cleaned up the comments.